### ISSUE TITLE
Setting Initial Resources

### ISSUE THREAD

**Issue by piosz** [kubernetes/kubernetes#13589]
_Warning: This issue suppressed the previous one #12149 since the previous one became messy due to the discussion about design started by me there._

This is an umbrella issue for tracking progress and discussing priorities of setting initial resources. Part of #10782.
##### Related docs

[proposals/initial-resources.md](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/initial-resources.md)
##### Sub-issues

The list is sorted by priorities. The features being blockers for 1.1 has appropriate tag.
- [x] [1.1] Initial implementation of InititalResources plugin #13132
- [ ] [1.1] data providers support
  - [x] support for InfluxDB #13132
  - [x] support for GCM #14113
  - [ ] support for Hawkular #13873
- [ ] [1.1] unit&e2e tests #14178
- [x] [1.1] annotation that resources have been set by the plugin #13985
- [ ] [1.1] documentation #15302
- [ ] [1.1] example
- [x] use namespace while making prediction #14972
- [ ] add events support
- [ ] prediction API
  - [ ] figure out how to provide Prediction API for users
  - [ ] define API
  - [ ] implement prediction API
- [ ] handle OOM
- [ ] add possibility to specify if estimation should be made

**Comment by piosz** [comment]
cc @fgrzadkowski @davidopp @jszczepkowski @mwielgus @kubernetes/goog-control-plane

**Comment by piosz** [comment]
@davidopp, @fgrzadkowski could you please take a look if the plan for 1.1 is reasonable for you?

**Comment by fgrzadkowski** [comment]
Assuming we should be code complete for 1.1 in ~3-4 weeks I'm worried you will not have time to do all of those things. In my opinion you should first focus on integrating with all backends so that this solutions works in all environments. Then work on providing more insight into the system (i.e. annotations) so that it's easy to debug. Then work on additional features in the following order:
1. prediction API
2. handle OOM
3. add ability to always default to hardcoded values from LimitRanger (but only once we know that people will need it; personally I think we don't)

Priority of 1) and 2) might change based on the input from RH about the importance of prediction API for UI.

**Comment by jszczepkowski** [comment]
@smarterclayton can you comment how "prediction API" is important and urgent for you?

**Comment by smarterclayton** [comment]
It's useful but not urgent - agree on the points Filip made.  I think we would prefer to have the useful core capabilities as represented by 1/2 working cleanly and we can expand into prediction.  @mwringe we should figure out who is going to jump on the hawkular support (who has time?).

**Comment by davidopp** [comment]
@vishh

**Comment by vishh** [comment]
+1 on @fgrzadkowski's points. Having a stable Metrics APIs that works across deployments is a pre-cursor for this feature.

**Comment by piosz** [comment]
Reassigning to @jszczepkowski and @krzysztofgrygiel who will work on the new version of Iniital Resources/Vertical Pod Autoscaler.

**Comment by mwielgus** [comment]
Closing in favor of Vertical Pod Autoscaler effort in kubernetes/autoscaler.
